### ISSUE TITLE
Add a human readable helper method for directory mappings

### ISSUE THREAD

**Issue by ivanfrain** [sbt/sbt-native-packager#161]
NOTE: pull request #159 follow up.

Currently, a directory mapping can be done in build.sbt with the following snippet (see #160 ): 

``` scala
mappings in Universal <++= (packageBin in Compile, target ) map { (_, target) =>
        val dir = target / "scala-2.10" / "api"
        (dir.***) pair relativeTo(dir.getParentFile)
} 
```

The idea behind this code is to put in the generated package an api directory with all its content. The original directory is located in target/scala-2.10/api. The target directory will be located at the pacakge root in the "api" directory.

A great enhancement would be to have a helper method simplifying the above snippet to something more comprehensive for human being, for example : 

``` scala
mappings in Universal ++= allFilesRelativeTo(file(target / "scala-2.10" / "api"))
```

**Comment by belgacea** [comment]
@muuki88 Is this feature abandoned?

**Comment by nigredo-tori** [comment]
@belgacea, I don't know if this was ever implemented in SNP itself, but if you're on SBT 1.0, `sbt.Path.directory` seems to do exactly this.

**Comment by muuki88** [comment]
We move some of those helpers to sbt-io itself and they added quite some more :)
